query/workmanager.go

```diff
@@ -321,11 +322,48 @@ Loop:
 				}
 
 			// If the query ended with any other error, put it back
-			// into the work queue.
+			// into the work queue if it has not reached the
+			// maximum number of retries.
 			case result.err != nil:
 				// Punish the peer for the failed query.
 				w.cfg.Ranking.Punish(result.peer.Addr())
```

Comment:
pre-existing but wondering if we should remove this Punish here and let the caller handle it instead since this could just be erroring cause we requested something that is not there anymore (like in a re-org)